### PR TITLE
Add easy @lines access and update runCallback

### PR DESCRIPTION
**Pull request by CoffeeFlux**

### CODE REVIEW

File: src/LineCollection.moon [hunk 1]
```diff
@@ -47,6 +47,38 @@ class LineCollection
...
+			else
+				indexKV[key] = value
+
+		setmetatable @, {
+			__index: newIndex
+			__newindex: newNewIndex
+			__len: =>
+				#@lines
+		}
+
```
Comment (torque) [inline src/LineCollection.moon]: I haven't fully tested it but I'm reasonably sure about 90% of the work being done here is unnecessary.
```moonscript
class IndexPasser
	new: =>
		meta = getmetatable @
		metaIdx = meta.__index

		meta.__index = ( index ) =>
			if 'number' == type index
				@lines[index]
			else
				metaIdx[index]

		meta.__newindex = ( index, value ) =>
			if 'number' == type index
				@lines[index] = value
			else
				rawset @, index, value

		@lines = { 5, 4, 3, 2, 1 }
```

**Comment by torque** [comment]
I'm just going to bump this to say that you never gave me a test case where my suggested implementation didn't work.

Also, I can't remember if we talked about this, but it would reduce the amount of nasty implementation to use the `__call` method instead. The only downside of this is that access and assignment would become asymmetrical, but that's not really my problem.

In the interest of convenience, it may also be worth it to implement `__pairs`. I think luajit in 5.2 compatibility mode supports this.

**Comment by torque** [comment]
you are extremely ugly and bad

**Comment by CoffeeFlux** [comment]
Commits are updated, not using call, and including pairs and ipairs. If you could reopen this that'd be lovely. Happy Valentine's Day!

**Pull request closed without merging**